src/ray/core_worker/context.cc

```diff
@@ -172,12 +178,32 @@ ObjectIDIndexType WorkerContext::GetNextPutIndex() {
   return GetThreadContext().GetNextPutIndex();
 }
 
+void WorkerContext::MayInitializeJobInfo(const JobID &job_id,
+                                         const rpc::JobConfig &job_config) {
+  absl::WriterMutexLock lock(&mutex_);
+  if (current_job_id_.IsNil()) {
+    current_job_id_ = job_id;
+  }
+  if (google::protobuf::util::MessageDifferencer::Equals(job_config_,
+                                                         kDefaultJobConfig)) {
+    job_config_ = job_config;
+  }
+}
+
 int64_t WorkerContext::GetTaskDepth() const {
   absl::ReaderMutexLock lock(&mutex_);
   return task_depth_;
 }
 
-const JobID &WorkerContext::GetCurrentJobID() const { return current_job_id_; }
+const JobID &WorkerContext::GetCurrentJobID() const {
```

Comment:
Should we make these functions return by-value instead of by-reference given that now access must be guarded by mutex?